src/test/java/org/opensearch/security/test/helper/rest/RestHelper.java

```diff
@@ -379,11 +379,27 @@ public String findValueInJson(final String jsonDotPath) {
 					fail("Invalid json dot path '" + jsonDotPath + "', rewrite with '.' characters between path elements.");
 				}
 				do {
-					final String pathEntry = jsonPathScanner.next();
+					String pathEntry = jsonPathScanner.next();
+					// if pathEntry is an array lookup
+					boolean isArrayLookup = pathEntry.contains("[");
+					int arrayEntryIdx = -1;
+					if (isArrayLookup) {
+						arrayEntryIdx = Integer.parseInt(pathEntry.replaceAll("[^0-9]", ""));
+						pathEntry = pathEntry.replaceAll("[^a-zA-Z_-]", "");
+					}
+
 					if (!currentNode.has(pathEntry)) {
 						fail("Unable to resolve '" + jsonDotPath + "', on path entry '" + pathEntry + "' from available fields " + currentNode.toPrettyString());
 					}
 					currentNode = currentNode.get(pathEntry);
+
+					// if it's an Array lookup we get the request index item
+					if (isArrayLookup) {
+						if(!currentNode.isArray()) {
+							fail("Request array lookup '" + jsonDotPath + "', but '" + pathEntry + "' field is not an array " + currentNode.toPrettyString());
```

Comment:
The failure message should explicitly states something like `"Unable to resolve '" + jsonDotPath + "', ......`

Reply:
done